contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,332 @@
+        Borrower storage borrower = borrowers[msg.sender];
+        uint256 depositedCollateralAfterRepay = _repay(msg.sender);
+
+        borrower.debt = 0;
+        borrower.collateralInSOHM = uint128(depositedCollateralAfterRepay);
+        borrower.collateralInGOHM = IgOHM(gOHM).balanceTo(depositedCollateralAfterRepay);
+    }
+
+    /**
+     * @notice repay debt with collateral and withdraw the accrued earnings to sOHM/gOHM
+     * - msg.sender must be a borrower
+     * - borrower must have outstanding debt
+     * @param _tokenToReceiveExcess amount of OHM to borrow
+     */
+    function repayDebtWithCollateralAndWithdrawTheRest(address _tokenToReceiveExcess) external {
+        uint256 depositedCollateralAfterRepay = _repay(msg.sender);
+        assignBorrowerInfoToZero(msg.sender);
+
+        if (depositedCollateralAfterRepay > 0) {
+            if (_tokenToReceiveExcess == sOHM) {
+                IERC20(sOHM).safeTransfer(msg.sender, depositedCollateralAfterRepay);
+            } else if (_tokenToReceiveExcess == gOHM) {
+                IERC20(sOHM).approve(staking, depositedCollateralAfterRepay);
+                IStaking(staking).wrap(msg.sender, depositedCollateralAfterRepay);
+            }
+        }
+    }
+
+    /**
+     * @notice deposits OHM to pay debt
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     * - borrower must have outstanding debt
+     * @param _ohmAmount amount of OHM to borrow
+     */
+    function repayDebtWithOHM(uint256 _ohmAmount) external isBorrower(msg.sender) {
+        if (borrowers[msg.sender].debt == 0) revert IncurDebtV1_BorrowerHasNoOutstandingDebet(msg.sender);
+        IERC20(OHM).safeTransferFrom(msg.sender, address(this), _ohmAmount);
+
+        totalOutstandingDebt -= _ohmAmount;
+        borrowers[msg.sender].debt = uint128(borrowers[msg.sender].debt - _ohmAmount);
+
+        IERC20(OHM).approve(treasury, _ohmAmount);
+        ITreasury(treasury).repayDebtWithOHM(_ohmAmount);
+    }
+
+    /**
+     * @notice repays debt using collateral and returns remaining tokens to borrower
+     * - onlyOwner (or governance)
+     * - sends remaining tokens to owner in sOHM
+     * @param _borrower the address that will interact with contract
+     * @param _to where to send remaining sOHM
+     */
+    function forceRepay(address _borrower, address _to) external onlyGovernor {
+        uint256 collateralAfterDebtPayment = _repay(_borrower);
+
+        assignBorrowerInfoToZero(_borrower);
+        if (collateralAfterDebtPayment > 0) IERC20(sOHM).safeTransfer(_to, collateralAfterDebtPayment);
+    }
+
+    /**
+     * @notice seize and burn _borrowers collateral and forgive debt
+     * - will burn all collateral, including excess of debt
+     * - onlyGovernance
+     * @param _borrower the account to seize
+     */
+    function seize(address _borrower) external onlyGovernor isBorrower(_borrower) {
```

Comment:
when will we want to use this function?

Reply:
My guess is when we have a bad borrower.

Reply:
Yep. This is a safeguard for any unforeseen reasons that may require Policy to kick out a borrower